samples/nrf_desktop/configuration/buttons.h

```diff
@@ -0,0 +1,55 @@
+struct button {
+	u8_t pin;
+	u8_t port;
+};
+
+#if CONFIG_BOARD_NRF52840_PCA20041
+const struct button col[] = {
+	{
+		.pin = 2,
+		.port = 0,
```

Comment:
I would like to keep an integer value `0` instead of `DT_` which translates to `"GPIO_0"`. This is because port number is used as index to an array of `gpio_devs` in every pin operation.

Reply:
I am against. It is more convenient for us but also this would be incompatible with the rest of the code. It would be better to simply define pins on port 1 as 32+pin_num (still incompatible with zephyr but using less memory and compatible with nrfx).

Reply:
Let's talk about it offline.

Reply:
I propose following:

```
#define PORT_COUNT 2
static const * const char port_map[PORT_COUNT] = {
   DT_GPIO_P0_DEV_NAME, DT_GPIO_P1_DEV_NAME}
};

const struct button col[] = {
   {0, n},
   {1, m},
...
};
```
then in source file map would be used only during obtaining gpio device pointer and later on we would use port id. Port count would also be defined in this header file.

Reply:
even better if there is no `PORT_COUNT` at all and in code you use `ARRAY_SIZE(port_map)`.